### ISSUE TITLE
archivers/php56-phar fails build: bad syntax in BUILD_DEPENDS

### ISSUE THREAD

**Issue by rwp0** [mariancerny/freebsd-ports-php56#7]
First of all, I'd like to thank you @mariancerny for maintaining this custom port
as there is still a lot of websites running on PHP 5.6 in production.

```
archivers/php56-phar # make
===>   php56-phar-5.6.40 depends on file: /usr/local/bin/phpize - found
===>   php56-phar-5.6.40 depends on package: autoconf>0 - found
Error: bad dependency syntax in BUILD_DEPENDS
expecting: pattern:origin[@flavour][:target]
got: /usr/local/lib/php/20131226/hash.so:
Errors with dependencies.
*** Error code 1

Stop.
make: stopped in archivers/php56-phar
```
Please help in resolving this; I need it to install `php56-extensions` meta-package, to `pkg create` a PHP 5.6 bundle.

The rest (55 packages) built nicely on [FreeBSD 13.1](https://www.freebsd.org/releases/13.1R/announce/) using

```
mount -t unionfs /opt/php56 /usr/ports
```

The default option of `-o rw` is needed rather than `-o ro` for creating work directories while building.

This is a list of packages that were built and installed successfully on `13.1-RELEASE`:

- php56-5.6.40
- php56-bcmath-5.6.40
- php56-bz2-5.6.40
- php56-calendar-5.6.40
- php56-ctype-5.6.40
- php56-curl-5.6.40
- php56-dba-5.6.40
- php56-dom-5.6.40
- php56-enchant-5.6.40
- php56-exif-5.6.40
- php56-fileinfo-5.6.40
- php56-filter-5.6.40
- php56-ftp-5.6.40
- php56-gd-5.6.40
- php56-gettext-5.6.40
- php56-gmp-5.6.40
- php56-iconv-5.6.40
- php56-imap-5.6.40
- php56-json-5.6.40
- php56-ldap-5.6.40
- php56-mbstring-5.6.40
- php56-mcrypt-5.6.40
- php56-mssql-5.6.40
- php56-mysql-5.6.40
- php56-mysqli-5.6.40
- php56-odbc-5.6.40
- php56-opcache-5.6.40
- php56-openssl-5.6.40
- php56-pcntl-5.6.40
- php56-pdo-5.6.40
- php56-pdo_dblib-5.6.40
- php56-pdo_firebird-5.6.40
- php56-pdo_mysql-5.6.40
- php56-pdo_odbc-5.6.40
- php56-pdo_pgsql-5.6.40_1
- php56-pdo_sqlite-5.6.40
- php56-pecl-intl-3.0.0_17
- php56-pgsql-5.6.40_1
- php56-posix-5.6.40
- php56-pspell-5.6.40
- php56-readline-5.6.40
- php56-session-5.6.40
- php56-shmop-5.6.40
- php56-simplexml-5.6.40
- php56-snmp-5.6.40
- php56-soap-5.6.40
- php56-sockets-5.6.40
- php56-sqlite3-5.6.40
- php56-sysvmsg-5.6.40
- php56-sysvsem-5.6.40
- php56-sysvshm-5.6.40
- php56-tidy-5.6.40
- php56-tokenizer-5.6.40
- php56-wddx-5.6.40
- php56-xml-5.6.40
- php56-xmlreader-5.6.40
- php56-xmlrpc-5.6.40
- php56-xmlwriter-5.6.40
- php56-xsl-5.6.40
- php56-zip-5.6.40
- php56-zlib-5.6.40

Used the supplied `print/freetype2` for building `php56-gd`.

**Comment by rwp0** [comment]
@mariancerny could you please see this?

Also (in addition to `php56-phar` port failing to build) the `php56-recode` package causes segmentation fault when loaded.

Thanks

**Comment by victor-sudakov** [comment]
I'm sorry to say that FreeBSD has probably become so irrelevant in the modern world of containerized apps that nobody cares any more.

**Comment by rwp0** [comment]
> I'm sorry to say that FreeBSD has probably become so irrelevant in the modern world of containerized apps that nobody cares any more.

Interesting point of view, though I don't agree with that.

PHP 5.6 on FreeBSD 13.1 is still being used and needed.

**Comment by victor-sudakov** [comment]
> PHP 5.6 on FreeBSD 13.1 is still being used and needed.

In Linux you can install and run multiple versions of PHP simultaneously (almost out of the box). Sorry for the rant but it was one of the major reasons for me to switch to Linux.

**Comment by mariancerny** [comment]
Sorry, I am not using PHP 5.6 any more (finally), so I am not maintaining freebsd-ports-php56 any more.

For the mount `-rw` option... This can be avoided by using the following statement in `make.conf`:

    WRKDIRPREFIX ?= /tmp/portbuild

**Comment by rwp0** [comment]
A note to self and others interested:

In `/opt/php56/archivers/php56-phar/work/php-5.6.40/ext/phar` where the extension's source lives
(ie. within the `work` directory of the port source) :

```make
In file included from /u/php56/archivers/php56-phar/work/php-5.6.40/ext/phar/util.c:23:
In file included from ./phar_internal.h:59:
In file included from /usr/local/include/php/ext/spl/spl_array.h:26:
In file included from /usr/local/include/php/ext/spl/spl_iterators.h:27:
/usr/local/include/php/ext/pcre/php_pcre.h:29:10: fatal error: 'pcre.h' file not found
#include "pcre.h"
         ^~~~~~~~
1 warning and 1 error generated.
*** Error code 1

Stop.
make: stopped in /opt/php56/archivers/php56-phar/work/php-5.6.40/ext/phar
```

Could **compile** and **install** `phar.so` by doing the following:

Edit `/usr/local/include/php/ext/pcre/php_pcre.h` (installed by `php56-5.6.40`):

```c
19 /* $Id$ */
20 
21 #ifndef PHP_PCRE_H
22 #define PHP_PCRE_H
23 
24 #if HAVE_PCRE || HAVE_BUNDLED_PCRE
25 
26 #if HAVE_BUNDLED_PCRE
27 #include "pcrelib/pcre.h"
28 #else
29 // #include "pcre.h"
30 #include "/usr/local/include/pcre.h"
31 #endif
32 
```
Change `#include "pcre.h"` to  `#include "/usr/local/include/pcre.h"`

`pcre.h` header is installed by the FreeBSD [pcre](https://www.freshports.org/devel/pcre/) package.

![image](https://user-images.githubusercontent.com/68062695/186681071-23f300a1-2238-4a3a-b801-87d5b28d8814.png)

Basically including the full path in the header since it has relate path for some reason.

---

`make clean` runs the following:

```sh
find . -name \*.gcno -o -name \*.gcda | xargs rm -f
find . -name \*.lo -o -name \*.o | xargs rm -f
find . -name \*.la -o -name \*.a | xargs rm -f
find . -name \*.so | xargs rm -f
find . -name .libs -a -type d|xargs rm -rf
rm -f libphp.la       modules/* libs/*
```
